src/utils/AddressUtils.ts

```diff
@@ -0,0 +1,18 @@
+import { providers } from "ethers";
+
+/**
+ * Checks if a contract is deployed at the given address
+ * @param address The ETH address to check
+ * @param provider A valid Ethers.js provider
+ * @returns A boolean indicating if a contract is deployed at the given address or not (true = contract, false = no contract)
+ */
+export async function isContractAddress(address: string, provider: providers.Provider): Promise<boolean> {
+  // A base case for if the address is null
+  if (!address) {
+    return false;
+  }
+  // Retrieve the code at the address
+  const code = await provider.getCode(address);
+  // If the code is not empty, then there is a contract at this address
+  return code !== "0x";
```

Comment:
This heuristic should get us whether or not a contract is deployed at this address